R/get-net-task.R

```diff
@@ -15,16 +15,27 @@
 #' ospsuite:::.getContainerTask()
 #'
 #' @keywords internal
-.getNetTask <- function(taskName) {
+.getNewNetTask <- function(taskName) {
   rClr::clrCallStatic("OSPSuite.R.Api", paste0("Get", taskName))
 }
 
-#' @rdname .getNetTask
+
+#' @title .getContainerTask
+#' @description
+#'
 #' @keywords internal
 .getContainerTask <- function() {
-  if (is.null(ospsuiteEnv$containerTask)) {
-    ospsuiteEnv$containerTask <- .getNetTask("ContainerTask")
-  }
+  .getNetTask("ContainerTask")
+}
 
-  ospsuiteEnv$containerTask
+#' @title .getNetTask
+#'
+#' @param taskName
+#'
+#' @keywords internal
+.getNetTask <- function(taskName) {
+  if (is.null(ospsuiteEnv[[taskName]])) {
+    ospsuiteEnv[[taskName]] <- .getNewNetTask(taskName)
```

Comment:
Let's define a separate cache for tasks. Otherwise, the user could have access to them via `getOSPSuiteSetting()` which is not optimal.

Reply:
I created a `tasksEnv` in latest commits.